lib/features/recurring_donations/overview/pages/recurring_donations_overview_page.dart

```diff
@@ -0,0 +1,213 @@
+                    ),
+                    width: double.infinity,
+                    height: size.height * 0.76,
+                    child: Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        Text(
+                          //TODO: POEditor
+                          'Recurring donations',
+                          style: Theme.of(context).textTheme.titleLarge,
+                        ),
+                        Container(
+                          width: double.infinity,
+                          height: 2,
+                          color: Colors.black,
+                        ),
+                        const SizedBox(height: 10),
+                        SizedBox(
+                          height: size.height * 0.69,
+                          child: SingleChildScrollView(
+                            child: Column(
+                              children: recurringDonations
+                                  .map(
+                                    (recurringDonation) =>
+                                        RecurringDonationItem(
+                                      recurringDonation: recurringDonation,
+                                      isExtended: selectedRecurringDonation ==
+                                          recurringDonation,
+                                      onTap: () {
+                                        setState(() {
+                                          selectedRecurringDonation =
+                                              selectedRecurringDonation ==
+                                                      recurringDonation
+                                                  ? null
+                                                  : recurringDonation;
+                                        });
+                                      },
+                                      onCancel: () {
+                                        showDialog<bool>(
+                                          context: context,
+                                          builder: (_) =>
+                                              CancelRecurringDonationConfirmationDialog(
+                                            recurringDonation:
+                                                recurringDonation,
+                                          ),
+                                        ).then((result) {
+                                          if (result != null &&
+                                              result == true) {
+                                            _fetchRecurringDonations();
+                                          }
+                                        });
+                                      },
+                                    ),
+                                  )
+                                  .toList(),
+                            ),
```

Comment:
Maybe this is nice to have it as a separate widget with its own state to select the donation and have a callback that would call the selected donation